packages/compass-field-store/src/stores/store.js

```diff
@@ -1,231 +0,0 @@
-        if (key === 'count') {
-          // counts add up
-          return isNumber(objectValue)
-            ? objectValue + sourceValue
-            : sourceValue;
-        }
-        if (key === 'type') {
-          // Avoid the merge of 'Array' with 'Array' case becoming
-          // an array with a single value, i.e. ['Array']
-          if (objectValue === sourceValue) {
-            return objectValue;
-          }
-          // arrays concatenate and de-duplicate
-          if (isString(objectValue)) {
-            return uniq([objectValue, sourceValue]);
-          }
-          return isArray(objectValue)
-            ? uniq(objectValue.concat(sourceValue))
-            : sourceValue;
-        }
-        // all other keys are handled as per default
-        return undefined;
-      }
-    );
-  };
-
-  /**
-   * Generate the flattened list of fields for the FieldStore.
-   *
-   * @param  {Object} fields       flattened list of fields
-   * @param  {Array} nestedFields  sub-fields of topLevelFields (if existing)
-   */
-  store._flattenedFields = (fields, nestedFields) => {
-    if (!nestedFields) {
-      return;
-    }
-
-    for (const field of nestedFields) {
-      const fieldPathId = getFlattenedPathIdString(field.path);
-
-      const existingField = fields[fieldPathId] || {};
-      const newField = pick(field, FIELDS);
-      fields[fieldPathId] = store._mergeFields(existingField, newField);
-
-      // recursively search arrays and subdocuments
-      for (const type of field.types) {
-        if (type.name === 'Document') {
-          // add nested sub-fields
-          store._flattenedFields(fields, type.fields);
-        }
-        if (type.name === 'Array') {
-          // add arrays of arrays or subdocuments
-          store._flattenedArray(fields, type.types);
-        }
-      }
-    }
-  };
-
-  /**
-   * Helper to recurse into the "types" of the mongodb-schema superstructure.
-   *
-   * @param {Object} fields      flattened list of fields to mutate
-   * @param {Array} nestedTypes  the "types" array currently being inspected
-   */
-  store._flattenedArray = (fields, nestedTypes) => {
-    // Arrays have no name, so can only recurse into arrays or subdocuments
-    for (const type of nestedTypes) {
-      if (type.name === 'Document') {
-        // recurse into nested sub-fields
-        store._flattenedFields(fields, type.fields);
-      }
-      if (type.name === 'Array') {
-        // recurse into nested arrays (again)
-        store._flattenedArray(fields, type.types);
-      }
-    }
-  };
-
-  /**
-   * Processes the fields in a format compatible with an autocompleter.
-   *
-   * @param {Object} fields - The fields.
-   *
-   * @returns {Array} The array of autocomplete metadata.
-   */
-  store._processFieldsForAutocomplete = (fields) => {
-    return Object.keys(fields).map((key) => {
-      const field =
-        key.indexOf('.') > -1 || key.indexOf(' ') > -1 ? `"${key}"` : key;
```

Comment:
The only change compared to the old field store schema parsing logic is removing this "escaping" logic. This doesn't cover all the cases of escaping a field key anyway and autocompleter handles this now, no need for field store to handle that